internal/tools/aws/acm_types.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/acm"
+	"github.com/aws/aws-sdk-go-v2/service/acm/types"
+	"github.com/mattermost/mattermost-cloud/model"
+)
+
+// ACMAPI represents the series of calls we require from the AWS SDK v2 ACM Client
+type ACMAPI interface {
+	ListCertificates(ctx context.Context, params *acm.ListCertificatesInput, optFns ...func(*acm.Options)) (*acm.ListCertificatesOutput, error)
+	ListTagsForCertificate(ctx context.Context, params *acm.ListTagsForCertificateInput, optFns ...func(*acm.Options)) (*acm.ListTagsForCertificateOutput, error)
+}
+
+// newCertificateFromACMCertificateSummary converts an ACM's certificate summary into our own certificate type
+func newCertificateFromACMCertificateSummary(c types.CertificateSummary) *model.Certificate {
+	return &model.Certificate{
+		ARN: c.CertificateArn,
```

Comment:
Is it possible that **ARN** could be nil?

Reply:
It's possible that `c.CertificateArn` would be nil ([docs](https://github.com/aws/aws-sdk-go-v2/blob/main/service/acm/types/types.go#L18)). 

Right now we search ACM Certs by the CertificateARN ([here](https://github.com/mattermost/mattermost-cloud/blob/048a9462589518fc7ae3066b3ab513814ee498f6/internal/tools/aws/acm.go#L33)) so it is supposed to return a certificate with an ARN.

Logic can be improved here but after thinking about it, since this is a transformation from an AWS Type to our own Type, I think the best way forward is to maintain the nullability in our types since we may require nullables in the future, so the method usages need to validate that on their end. (Just added it in https://github.com/mattermost/mattermost-cloud/pull/764/commits/f78ea8158bee5e2dabf67b0abc78f665797531b1, thanks for noticing)